src/sleep/sleep.c

```diff
@@ -275,28 +275,26 @@ static int custom_timer_suspend(const SleepConfig *sleep_config) {
         while (battery_is_low() == 0) {
                 _cleanup_close_ int tfd = -1;
                 struct itimerspec ts = {};
-                usec_t suspend_interval = sleep_config->hibernate_delay_usec, before_timestamp = 0, after_timestamp = 0, total_suspend_interval;
+                usec_t suspend_interval, before_timestamp, after_timestamp;
                 bool woken_by_timer;
 
                 tfd = timerfd_create(CLOCK_BOOTTIME_ALARM, TFD_NONBLOCK|TFD_CLOEXEC);
                 if (tfd < 0)
                         return log_error_errno(errno, "Error creating timerfd: %m");
 
-                /* Store current battery capacity and current time before suspension */
+                /* Store current battery capacity before suspension */
                 r = fetch_batteries_capacity_by_name(&last_capacity);
-                if (r >= 0)
-                        before_timestamp = now(CLOCK_BOOTTIME);
-                else if (r == -ENOENT)
```

Comment:
don't remove this, or no battery becomes a hard error, which is wrong

Reply:
`fetch_batteries_capacity_by_name()` does not return -ENOENT.

Reply:
I'm pretty sure it did at some point, and that's why it was added. It might again in the future after the next refactoring if it doesn't anymore now, so just leave it, it won't hurt

Reply:
We usually do not keep unused code. Let's reintroduce when necessary.
Note, I am planning more cleanups, but not included in this PR, as this is already big enough.